mne/channels/channels.py

```diff
@@ -890,6 +896,20 @@ def _pick_drop_channels(self, idx):
             assert isinstance(self, BaseRaw) and not self.preload
         return self
 
+    def _pick_projs(self):
+        """Only keep projectors whose channels are still in the data."""
+        drop_idx = []
+        for idx, proj in enumerate(self.info['projs']):
+            if not set(self.info['ch_names']) & set(proj['data']['col_names']):
+                drop_idx.append(idx)
+
+        for idx in drop_idx:
+            logger.warning(f"Removing projector {self.info['projs'][idx]}")
```

Comment:
don't use `logger.warning` but rather `warn` from `..utils`

Reply:
Weird, ok… why is that?

Reply:
We always use `warn`. It does nice things like sets `stacklevel` automatically, and does or does not emit based on the `logger.level`, etc.

Reply:
Thanks, that's good to know!